### ISSUE TITLE
settings-local.yaml not working

### ISSUE THREAD

**Issue by Exw27** [imartinez/privateGPT#1108]
Traceback (most recent call last):
  File "/home/exw/.cache/pypoetry/virtualenvs/private-gpt-gMd4n001-py3.11/lib/python3.11/site-packages/injector/__init__.py", line 798, in get
    return self._context[key]
           ~~~~~~~~~~~~~^^^^^
KeyError: <class 'private_gpt.server.ingest.ingest_service.IngestService'>

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/exw/.cache/pypoetry/virtualenvs/private-gpt-gMd4n001-py3.11/lib/python3.11/site-packages/injector/__init__.py", line 798, in get
    return self._context[key]
           ~~~~~~~~~~~~~^^^^^
KeyError: <class 'private_gpt.components.embedding.embedding_component.EmbeddingComponent'>

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "<frozen runpy>", line 198, in _run_module_as_main
  File "<frozen runpy>", line 88, in _run_code
  File "/home/exw/Repos/privateGPT/private_gpt/__main__.py", line 5, in <module>
    from private_gpt.main import app
  File "/home/exw/Repos/privateGPT/private_gpt/main.py", line 123, in <module>
    from private_gpt.ui.ui import mount_in_app
  File "/home/exw/Repos/privateGPT/private_gpt/ui/ui.py", line 19, in <module>
    ingest_service = root_injector.get(IngestService)
                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/exw/.cache/pypoetry/virtualenvs/private-gpt-gMd4n001-py3.11/lib/python3.11/site-packages/injector/__init__.py", line 91, in wrapper
    return function(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/exw/.cache/pypoetry/virtualenvs/private-gpt-gMd4n001-py3.11/lib/python3.11/site-packages/injector/__init__.py", line 974, in get
    provider_instance = scope_instance.get(interface, binding.provider)
                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/exw/.cache/pypoetry/virtualenvs/private-gpt-gMd4n001-py3.11/lib/python3.11/site-packages/injector/__init__.py", line 91, in wrapper
    return function(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/exw/.cache/pypoetry/virtualenvs/private-gpt-gMd4n001-py3.11/lib/python3.11/site-packages/injector/__init__.py", line 800, in get
    instance = self._get_instance(key, provider, self.injector)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/exw/.cache/pypoetry/virtualenvs/private-gpt-gMd4n001-py3.11/lib/python3.11/site-packages/injector/__init__.py", line 811, in _get_instance
    return provider.get(injector)
           ^^^^^^^^^^^^^^^^^^^^^^
  File "/home/exw/.cache/pypoetry/virtualenvs/private-gpt-gMd4n001-py3.11/lib/python3.11/site-packages/injector/__init__.py", line 264, in get
    return injector.create_object(self._cls)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/exw/.cache/pypoetry/virtualenvs/private-gpt-gMd4n001-py3.11/lib/python3.11/site-packages/injector/__init__.py", line 998, in create_object
    self.call_with_injection(init, self_=instance, kwargs=additional_kwargs)
  File "/home/exw/.cache/pypoetry/virtualenvs/private-gpt-gMd4n001-py3.11/lib/python3.11/site-packages/injector/__init__.py", line 1031, in call_with_injection
    dependencies = self.args_to_inject(
                   ^^^^^^^^^^^^^^^^^^^^
  File "/home/exw/.cache/pypoetry/virtualenvs/private-gpt-gMd4n001-py3.11/lib/python3.11/site-packages/injector/__init__.py", line 91, in wrapper
    return function(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/exw/.cache/pypoetry/virtualenvs/private-gpt-gMd4n001-py3.11/lib/python3.11/site-packages/injector/__init__.py", line 1079, in args_to_inject
    instance: Any = self.get(interface)
                    ^^^^^^^^^^^^^^^^^^^
  File "/home/exw/.cache/pypoetry/virtualenvs/private-gpt-gMd4n001-py3.11/lib/python3.11/site-packages/injector/__init__.py", line 91, in wrapper
    return function(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/exw/.cache/pypoetry/virtualenvs/private-gpt-gMd4n001-py3.11/lib/python3.11/site-packages/injector/__init__.py", line 974, in get
    provider_instance = scope_instance.get(interface, binding.provider)
                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/exw/.cache/pypoetry/virtualenvs/private-gpt-gMd4n001-py3.11/lib/python3.11/site-packages/injector/__init__.py", line 91, in wrapper
    return function(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/exw/.cache/pypoetry/virtualenvs/private-gpt-gMd4n001-py3.11/lib/python3.11/site-packages/injector/__init__.py", line 800, in get
    instance = self._get_instance(key, provider, self.injector)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/exw/.cache/pypoetry/virtualenvs/private-gpt-gMd4n001-py3.11/lib/python3.11/site-packages/injector/__init__.py", line 811, in _get_instance
    return provider.get(injector)
           ^^^^^^^^^^^^^^^^^^^^^^
  File "/home/exw/.cache/pypoetry/virtualenvs/private-gpt-gMd4n001-py3.11/lib/python3.11/site-packages/injector/__init__.py", line 264, in get
    return injector.create_object(self._cls)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/exw/.cache/pypoetry/virtualenvs/private-gpt-gMd4n001-py3.11/lib/python3.11/site-packages/injector/__init__.py", line 998, in create_object
    self.call_with_injection(init, self_=instance, kwargs=additional_kwargs)
  File "/home/exw/.cache/pypoetry/virtualenvs/private-gpt-gMd4n001-py3.11/lib/python3.11/site-packages/injector/__init__.py", line 1040, in call_with_injection
    return callable(*full_args, **dependencies)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/exw/Repos/privateGPT/private_gpt/components/embedding/embedding_component.py", line 19, in __init__
    self.embedding_model = HuggingFaceEmbedding(
                           ^^^^^^^^^^^^^^^^^^^^^
  File "/home/exw/.cache/pypoetry/virtualenvs/private-gpt-gMd4n001-py3.11/lib/python3.11/site-packages/llama_index/embeddings/huggingface.py", line 58, in __init__
    import torch
  File "/home/exw/.cache/pypoetry/virtualenvs/private-gpt-gMd4n001-py3.11/lib/python3.11/site-packages/torch/__init__.py", line 235, in <module>
    from torch._C import *  # noqa: F403
    ^^^^^^^^^^^^^^^^^^^^^^
ImportError: libcudnn.so.8: cannot open shared object file: No such file or directory

**Comment by imartinez** [comment]
The error comes from your CUDA installation, not privateGPT itself. 

I found these two related issues, it may help you solve the problem:
https://github.com/pytorch/pytorch/issues/104259
https://github.com/NVIDIA/TensorRT/issues/1747

Please let us know if you managed to solve it and how, so we can improve the troubleshooting section in the docs. 

I'll leave this issue open temporarily so we can have visibility on the fix process. But just to be clear, given it is a specific setup issue (with torch, C, CUDA), PrivateGPT won't be actively looking into it.

**Comment by Exw27** [comment]
Fixed it by running `poetry add torch==2.0.0` It is a torch issue related to torch==2.0.1 so specifying torch==2.0.0 fixes the issue.
Thanks for the help.
